### PR TITLE
Don't reload page after saving addon settings in a popup

### PR DESCRIPTION
**Pull request by beckyvb**
Add a flag to let the modal know not to refresh after form save and apply it to the addon settings.
Also clean js a bit.

Closes https://github.com/vanilla/vanilla/issues/4476

### CODE REVIEW

File: applications/dashboard/js/dashboard.js [hunk 1]
```diff
@@ -345,9 +345,8 @@ var DashboardModal = (function() {
 
         defaultSettings: {
             httpmethod: 'get',
-            afterSuccess: function(json, sender) {
-                // Called after the confirm url has been loaded via ajax
-            },
+            afterSuccess: function(json, sender) {},
```
Comment (DaazKu) [inline applications/dashboard/js/dashboard.js]: You could just use `$.noop` there.

File: applications/dashboard/js/src/modal.dashboard.js [hunk 2]
```diff
@@ -22,9 +22,8 @@ var DashboardModal = (function() {
 
         defaultSettings: {
             httpmethod: 'get',
-            afterSuccess: function(json, sender) {
-                // Called after the confirm url has been loaded via ajax
-            },
+            afterSuccess: function(json, sender) {},
```
Comment (DaazKu) [inline applications/dashboard/js/src/modal.dashboard.js]: You could just use $.noop there.

Comment (beckyvb) [inline applications/dashboard/js/src/modal.dashboard.js]: This gives a developer a better clue of how to override the setting.

Comment (DaazKu) [inline applications/dashboard/js/src/modal.dashboard.js]:
```js
$.noop, // function(json, sender)
```

Save a useless function declaration. It is no big deal though. Just something to keep in mind as a good practice.

**Pull request merged**